src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -51,66 +51,79 @@ export class ReactNativeErrorHandlers implements Integration {
+          .default;
+      } catch (e) {
+        // do nothing, as the user may be running an older version of RN without promiseRejectionTrackingOptions
+      }
+
       tracking.disable();
       tracking.enable({
         allRejections: true,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        onUnhandled: (id: any, error: any) => {
+        onUnhandled: (id: string, error: Error) => {
           if (__DEV__) {
-            // We mimic the behavior of unhandled promise rejections showing up as a warning.
-            // eslint-disable-next-line no-console
-            console.warn(id, error);
+            promiseRejectionTrackingOptions.onUnhandled(id, error);
           }
+
           getCurrentHub().captureException(error, {
             data: { id },
             originalException: error,
           });
         },
+        onHandled: (id: string) => {
+          promiseRejectionTrackingOptions.onHandled(id);
+        },
       });
-
-      /* eslint-disable
-        @typescript-eslint/no-var-requires,
-        import/no-extraneous-dependencies,
-        @typescript-eslint/no-explicit-any,
-        @typescript-eslint/no-unsafe-member-access
-      */
+    }
+  }
+  /**
+   * Checks if the promise is the same one or not, if not it will warn the user
+   */
+  private _checkPromiseVersion(): void {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires,import/no-extraneous-dependencies
       const Promise = require("promise/setimmediate/core");
-      const _global = getGlobalObject<any>();
 
-      /* In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
-        Due to this, we need to take the methods that tracking.enable sets, and then set them on the global promise.
-        Note: We do not want to overwrite the whole promise in case there are extensions present.
+      const _global = getGlobalObject<{ Promise: typeof Promise }>();
 
-        If the global promise is the same as the imported promise (expected in RN <0.63), we do nothing.
-      */
-      const _onHandle = Promise._onHandle ?? Promise._Y;
-      const _onReject = Promise._onReject ?? Promise._Z;
-
-      if (
-        Promise !== _global.Promise &&
-        typeof _onHandle !== "undefined" &&
-        typeof _onReject !== "undefined"
-      ) {
-        if ("_onHandle" in _global.Promise && "_onReject" in _global.Promise) {
-          _global.Promise._onHandle = _onHandle;
-          _global.Promise._onReject = _onReject;
-        } else if ("_Y" in _global.Promise && "_Z" in _global.Promise) {
-          _global.Promise._Y = _onHandle;
-          _global.Promise._Z = _onReject;
-        }
+      if (Promise !== _global.Promise) {
+        logger.warn(
+          "Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page."
```

Comment:
link to the docs once merged

Reply:
@marandaneto Do we want to directly have a link to the docs in the warning message?

Reply:
since it mentions the troubleshooting page its fine the way how it is